Small fixed error in code

DarkBullNull opened the pull request:
If define CALL_ARG_STATS = 1 and DISPLAY_SIZES = 1 in "jit.h", then will be compilation errors 

- Remove unused pointers
- Fixed bool-check error call
- Fixed the type declaration for dataSize
- Grammar fix in "compiler.h"

DarkBullNull committed:
Fix bool-check call and add nullptr

DarkBullNull committed:
Add lvar dataSize

If def DISPLAY_SIZES = 1 in jit.h, then gives and error

DarkBullNull committed:
Remove type def

If def DISPLAY_SIZES = 1, gives an error

dnfadmin commented:
[![CLA assistant check](https://cla.dotnetfoundation.org/pull/badge/not_signed)](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=51821) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=51821) before we can accept your contribution.<br/><br/>:x: DarkBullNull [sign now](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=51821)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/runtime?pullRequest=51821) it.</sub>

SingleAccretion commented on src/coreclr/jit/codegen.h:
```diff
@@ -208,6 +208,7 @@ class CodeGen final : public CodeGenInterface
     void**    codePtr;
     uint32_t* nativeSizeOfCode;
     unsigned  codeSize;
+    size_t    dataSize;
```
This should probably live with the other related fields in [compiler.h](https://github.com/dotnet/runtime/blob/79d8dadf915d7d13b0aee3933c5b3d51a4940159/src/coreclr/jit/compiler.h#L11226).

DarkBullNull replied:
Hm, okay, i moved define dataSize in compiler.h
Thank you!

DarkBullNull committed:
Move def dataSize

DarkBullNull committed:
Update codegen.h

DarkBullNull committed:
Update compiler.h

DarkBullNull committed:
Rename "dataSize" to "eDataSize" (EmitDataSize)

because of intersection of names "dataSize" between "compiler.h" and in "codegenarm64.cpp"

DarkBullNull committed:
Grammar fix compiler.h

DarkBullNull committed:
Grammar fix

DarkBullNull commented:
@BruceForstall

sandreenko commented on src/coreclr/jit/gentree.cpp:
```diff
@@ -16564,7 +16564,7 @@ bool GenTree::IsPhiDefn()
 //
 // Return Value:
-//    Returns "true" iff 'this' is a GT_LCL_FLD or GT_STORE_LCL_FLD on which the type
+//    Returns "true" if 'this' is a GT_LCL_FLD or GT_STORE_LCL_FLD on which the type
```
iff is widely used in the jit code base as "if and only if"

sandreenko commented on src/coreclr/jit/compiler.cpp:
```diff
@@ -7451,8 +7452,8 @@ void Compiler::compJitStats()
 {
-    GenTree* args;
-    GenTree* argx;
+    GenTree* args = nullptr;
```
where is 'args' and 'argx' used in this method? Looks like they can be just deleted.

DarkBullNull replied:
Thank you for reviewing! I've fixed this all.

sandreenko reviewed (commented):
> Fixed the type declaration for dataSize(because of 2523-2524 line doesn't see the declaration of this variable in "codegencommon.cpp"). Correct me, if i'm wrong

Is there an issue if we move 
```
size_t dataSize = GetEmitter()->emitDataSize();
```
from `genEmitMachineCode` to the next function instead of declaring it globally?

and revert eDataSize after.

DarkBullNull committed:
update codegencommon.cpp

transfer local declaration for dataSize

DarkBullNull committed:
Delete codegencommon.cpp

DarkBullNull committed:
Add files via upload

DarkBullNull committed:
Merge branch 'main' into main

DarkBullNull committed:
Update compiler.cpp

DarkBullNull committed:
eDataSize -> dataSize back

DarkBullNull committed:
Update gentree.cpp

DarkBullNull committed:
Update compiler.cpp

DarkBullNull committed:
Update codegencommon.cpp

DarkBullNull committed:
Update compiler.h

sandreenko commented on src/coreclr/jit/codegencommon.cpp:
```diff
@@ -2377,11 +2377,7 @@ void CodeGen::genEmitMachineCode()
     compiler->unwindReserve();
 
-#if DISPLAY_SIZES
 
```
the format job does not like these 3 empty lines, could you please delete 2?
You can fix it manually or apply the patch produced by the ci job (https://dev.azure.com/dnceng/_apis/resources/Containers/6976659/format.windows.x64.patch?itemPath=format.windows.x64.patch%2Fformat.patch) or run the format job locally ([doc](https://github.com/dotnet/jitutils/blob/main/doc/formatting.md)):
1. download https://github.com/dotnet/jitutils;
2. run bootstrap;
3. run `jit-format.bat --fix --coreclr path_to_runtime\runtime\src\coreclr`

DarkBullNull replied:
Thank you for your comment. Fixed it!

sandreenko reviewed (approved):
LGTM, thank you for restoring `CALL_ARG_STATS` and `DISPLAY_SIZES `.

sandreenko commented:
cc @dotnet/jit-contrib

DarkBullNull committed:
Update codegencommon.cpp

sandreenko commented:
The failures are unrelated, covered by https://statics.teams.cdn.office.net/evergreen-assets/safelinks/1/atp-safelinks.html
